### ISSUE TITLE
CAS number disclosure and proprietary data

### ISSUE THREAD

**Issue by NMeyer1357** [HPDCollaborative/builder-ui-beta-reporting#47]
If you manually add a substance by CAS number and then rename substance and select "I do not want to disclose this substance's CAS RN #", it will still show the screened substance name - making it impossible to enter materials proprietary to the HPD owner (as a consultant)

**Comment by 19peaches** [comment]
This cannot be addressed since the proper requested Bug Report information wasn't submitted.

**Comment by NMeyer1357** [comment]
Sorry!

 **Describe the bug**
When entering a proprietary CAS# material, when you rename the substance in order to protect its identity, the builder automatically puts the name of the substance back into the HPD and puts the renaming in parenthesis.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to add substance
2. Select add substance to manually enter substance
3. Enter CAS # and select (the name should pop up in the substance name field)
4. Rename substance by deleting substance name and entering “Proprietary Material”
5. Fill out remaining necessary fields and finalize by selecting add substance.
6. Error is observed as substance name appears as “Polyvinyl Chloride (Proprietary Material)” with the proprietary substance name listed 

**Expected behavior**
The substance should be added with the respective hazards of the CAS # inputted in step 3, with the name of the material appearing as “Proprietary Material”. However even when the substance name is deleted in step 4, the builder uses the substance name on the HPD and puts the renaming in parenthesis next to it, making it impossible to enter materials proprietary to the builder owner.

**Screenshots**
The builder 2.1.1 is down now so unfortunately I cannot reproduce screenshots.

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro
 - Chrome 64-bit Version 68.0.3440.106
 - Version 1709

**Additional context**
The above procedure steps 1-5 was the proper procedure for entering proprietary information to the manufacturer as a consultant in builder v2.1.

**Comment by 19peaches** [comment]
Great, that's much more useful.

Can you tell me the name of the product, or product id so that I can research this? I'd like to look in the database and see if something was stored incorrectly.

**Comment by 19peaches** [comment]
Alright I just tested this ... and although the substance name does show up in the UI, obviously or else you wouldn't have any idea what it is to edit it, it does not show up on the actual HPD as you can see in the screen below.

![screen shot 2018-08-28 at 5 13 58 am](https://user-images.githubusercontent.com/1456472/44722252-5b0ecd80-aa81-11e8-9f4b-43727de3d161.png)

Perhaps something was filled out incorrectly on the form.

**Comment by 19peaches** [comment]
@NMeyer1357 

Any further information on this? The staging app is NOT down.

Thanks.

**Comment by NMeyer1357** [comment]
I was having issues getting into the staging app this morning, but it works now, here's the full process to recreate and some more details:

1.	Add Substance
 
![image](https://user-images.githubusercontent.com/42770883/44739873-ff673300-aac6-11e8-854f-a4c6443f92d5.png)

2.	Add Manually
 
![image](https://user-images.githubusercontent.com/42770883/44739882-05f5aa80-aac7-11e8-84cd-ce515804d524.png)

3.	Enter CAS # (substance name auto-populates)
 
![image](https://user-images.githubusercontent.com/42770883/44739887-09893180-aac7-11e8-9b98-9f46a6088c6c.png)

4.	Delete substance name and enter alternate name “Proprietary”
 
![image](https://user-images.githubusercontent.com/42770883/44739893-0beb8b80-aac7-11e8-8494-75085a012831.png)

5.	Select “I do not want to disclose this substance’s CAS RN #”
 
![image](https://user-images.githubusercontent.com/42770883/44739896-0ee67c00-aac7-11e8-8cce-e026583a3b31.png)

6.	Fill out the remaining information and save substance
 
![image](https://user-images.githubusercontent.com/42770883/44739903-127a0300-aac7-11e8-9887-2053dea49a25.png)

7.	The substance comes up named in the account “POLYVINYL CHLORIDE (PVC) (Proprietary)”, where it should only come up as “Proprietary”
 
![image](https://user-images.githubusercontent.com/42770883/44739909-1574f380-aac7-11e8-8e83-656c4d3ccaa7.png)

I am aware of the feature where you can check “Do not disclose the name of this substance on the HPD”, however I need this feature as a consultant when entering substances proprietary to their suppliers in the manufacturer’s HPD builder account (which they grant us access to). When manually adding a substance and selecting “I do not want to disclose this substance’s CAS RN #”, you cannot go back in edit substance and see the CAS RN, while the substance remains screened. Since the builder automatically adds the chemical name to the substance name, and just adds the user inputted name (step 4 above) after the chemical name in the parenthesis, it is impossible to enter in proprietary data to the manufacturer, without them being able to have access to the chemical name of the substance. What used to happen in Builder 2.1 is that the substance would be renamed based on the user input (step 4), and it would appear on the substance list as whatever name was entered in step 4, without automatically adding the chemical name. This allowed for the entering of proprietary information and is a very necessary feature of the builder.
![image](https://user-images.githubusercontent.com/42770883/44739917-1a39a780-aac7-11e8-997f-2401d10e1731.png)
![image](https://user-images.githubusercontent.com/42770883/44739922-1c9c0180-aac7-11e8-916c-e57e1dd2d045.png)

**Comment by Millali** [comment]
Hi  Vince, 

This is correct, the original name and CAS number of proprietary substances should not display anywhere once the substance has been added. This was working at some point. 

Please check the record below as it indicates when the CAS and or name have been hidden. If you click to edit the substance you will see that neither the name nor the CAS are displayed. 

https://staging.hpd-collaborative.org/actions/2371/record

I am also attaching a screenshot of the edit substance screen. 
![screencapture-staging-hpd-collaborative-org-actions-2371-record-2018-08-28-13_47_20](https://user-images.githubusercontent.com/34175076/44740626-ef505300-aac8-11e8-9a14-82646ec97158.png)

Let me know if you need a live demo!

Milla

**Comment by 19peaches** [comment]
@Millali 

This never worked in the current version, as you can see by the created dates on your example substances, those were created in the current builder, not the new one.

Let me look and see how this is handled in the current builder, changing it on the form page shouldn't be an issue, but the table view draws directly from the name column in the db, I can't edit that as it's a pre-built third party component, all we can do is pass data in, not format how it displays.

Let me look into this further, perhaps this can be flagged in the backend to change the name ... that might work.

**Comment by 19peaches** [comment]
@Millali @NMeyer1357 

Ok, I've discovered what the issue is.

In the current builder, we have 2 different naming conventions.

If you select a substance from the initial typeahead, then rename this substance, this si where you get the name that displays both the original, and the parentheses. 

If you enter your substance manually and change the name, the original name is replaced altogether.

In the 2.1.1 builder we've removed the extra name saving step so as to reduce the overhead and complexity of the interface, so in essence, we only have a single name saving convention.

I will add a flag to the post routine so that the manually created name will forego the parentheses saving method and simply use whatever name is posted.

I think this will resolve the issue in both places, both the UI table display, and on the form.
